### ISSUE TITLE
Custom picker sometimes crashing when marking many pictures as "not for upload" towards the bottom

### ISSUE THREAD

**Issue by nicolas-raoul** [commons-app/apps-android-commons#5628]
Introduced by pull request https://github.com/commons-app/apps-android-commons/pull/5576

**Comment by shashankiitbhu** [comment]
@nicolas-raoul As discussed in the mentioned PR discussion I'll be solving this in a separate PR, so please assign this to me , thanks

**Comment by nicolas-raoul** [comment]
If I remember correctly, this one was happening with "Show already actioned pictures" disabled. I will try and update here.

**Comment by shashankiitbhu** [comment]
@nicolas-raoul I got the following error after crash:

```
ACRA caught a ConcurrentModificationException for fr.free.nrw.commons.beta
                                                                                                    java.util.ConcurrentModificationException
                                                                                                    	at java.util.ArrayList$Itr.checkForComodification(ArrayList.java:1029)
                                                                                                    	at java.util.ArrayList$Itr.next(ArrayList.java:982)
                                                                                                    	at fr.free.nrw.commons.customselector.ui.selector.CustomSelectorActivity$insertIntoNotForUpload$1.invokeSuspend(CustomSelectorActivity.kt:506)
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
                                                                                                    	at android.os.Handler.handleCallback(Handler.java:942)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:210)
                                                                                                    	at android.os.Looper.loop(Looper.java:299)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8252)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:559)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:954)
                                                                                                    	
                                                                                                    
```
Which suggests that the crash was due to a `ConcurrentModificationException` , which occurred when attempting to modify a collection concurrently with iterating over it based on this I have made some changes, and I think that fixes the issue.

**Comment by kanahia1** [comment]
Hey @shashankiitbhu, I was also facing same error in https://github.com/commons-app/apps-android-commons/pull/5635.

**Comment by shashankiitbhu** [comment]
@kanahia1 yes and I think that was the issue that was causing this error and that's why I am unable to reproduce this error after syncing with main.

@nicolas-raoul I think this issue is also Fixed by #5639 and can be closed, thanks
